From b694fc8fbe3bebbfb278cea3f6b9ccc8ed196124 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Tue, 21 Oct 2025 22:45:22 -0400
Subject: [PATCH 4/7] Fedora: attester: pick attesters in all-attesters
Content-type: text/plain

Currently: snp, tdx, az-snp-vtpm and az-tdx-vtpm

Remove all dependencies of non-built attesters

Some require additional dependencies that are not in Fedora

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 attestation-agent/attester/Cargo.toml | 34 +++++++--------------------
 1 file changed, 9 insertions(+), 25 deletions(-)

diff --git a/attestation-agent/attester/Cargo.toml b/attestation-agent/attester/Cargo.toml
index 5827f79..92d1a7d 100644
--- a/attestation-agent/attester/Cargo.toml
+++ b/attestation-agent/attester/Cargo.toml
@@ -15,8 +15,6 @@ az-snp-vtpm = { version = "0.8.1", default-features = false, features = [
 az-tdx-vtpm = { version = "0.8.1", default-features = false, features = [
     "attester",
 ], optional = true }
-azure-guest-attestation-sdk = { git = "https://github.com/Azure/azure-guest-attestation-sdk", rev = "9c8344cbeeb2c7ab46c1311e983f254a69ab5eec", optional = true }
-azure-tpm = { git = "https://github.com/Azure/azure-guest-attestation-sdk", rev = "9c8344cbeeb2c7ab46c1311e983f254a69ab5eec", optional = true }
 base64.workspace = true
 clap = { workspace = true, features = ["derive"], optional = true }
 cfg-if.workspace = true
@@ -24,9 +22,7 @@ crypto.path = "../deps/crypto"
 hex.workspace = true
 iocuddle = { version = "0.3.0", optional = true }
 kbs-types.workspace = true
-occlum_dcap = { git = "https://github.com/occlum/occlum", tag = "v0.29.7", optional = true }
 pem-rfc7468 = { version = "1", optional = true, features = ["alloc"] }
-pv = { version = "0.12.0", package = "s390_pv", optional = true }
 scroll = { version = "0.13.0", default-features = false, features = [
     "derive",
     "std",
@@ -39,12 +35,7 @@ sev = { version = "6.3.1", default-features = false, features = [
 ], optional = true }
 sha2.workspace = true
 strum.workspace = true
-tdx-attest-rs = { git = "https://github.com/intel/confidential-computing.tee.dcap", tag = "DCAP_1.27.101", optional = true }
 thiserror.workspace = true
-csv-rs = { version = "0.1.0", optional = true }
-codicon = { version = "3.0", optional = true }
-hyper = { version = "0.14", features = ["full"], optional = true }
-hyper-tls = { version = "0.5", optional = true }
 tokio = { workspace = true, features = ["rt", "macros", "fs", "io-util"] }
 tempfile = { workspace = true, optional = true }
 tracing.workspace = true
@@ -61,12 +52,6 @@ zerocopy = { version = "0.8.54", optional = true, features = ["derive"] }
 tokio.workspace = true
 rstest.workspace = true
 
-[target.'cfg(target_arch = "x86_64")'.dependencies]
-# When bumping this tag, check the crate's release notes on GitHub for its
-# corresponding libnvat/libnvat-dev version, and update the `version` input
-# default in .github/actions/install-nvat-sdk/action.yml to match.
-nv-attestation-sdk = { git = "https://github.com/NVIDIA/attestation-sdk", tag = "2026.06.09", optional = true }
-
 [[bin]]
 name = "evidence_getter"
 required-features = ["bin"]
@@ -76,16 +61,7 @@ default = ["all-attesters"]
 
 all-attesters = [
     "tdx-attester",
-    "sgx-attester",
-    "az-snp-vtpm-attester",
-    "az-tdx-vtpm-attester",
     "snp-attester",
-    "csv-attester",
-    "hygon-dcu-attester",
-    "cca-attester",
-    "se-attester",
-    "tpm-attester",
-    "nvidia-attester",
     "nvidia-dpu-attester",
 ]
 
@@ -94,24 +70,19 @@ all-attesters = [
 # generate TDX quotes with upstream kernels.
 tsm-report = ["tempfile"]
 tdx-attester = ["scroll", "tsm-report", "iocuddle", "zerocopy"]
-tdx-attest-dcap-ioctls = ["tdx-attest-rs"]
-sgx-attester = ["occlum_dcap"]
-nvidia-attester = ["nv-attestation-sdk"]
+tdx-attest-dcap-ioctls = []
+sgx-attester = []
+nvidia-attester = []
 nvidia-dpu-attester = ["p384"]
-az-snp-vtpm-attester = ["az-snp-vtpm", "pem-rfc7468", "azure-tpm"]
-az-tdx-vtpm-attester = [
-    "az-snp-vtpm-attester",
-    "az-tdx-vtpm",
-    "azure-guest-attestation-sdk",
-    "zerocopy",
-]
+az-snp-vtpm-attester = []
+az-tdx-vtpm-attester = []
 snp-attester = ["sev", "tsm-report"]
-csv-attester = ["hygon", "codicon", "hyper", "hyper-tls"]
-hygon-dcu-attester = ["hygon"]
-hygon = ["csv-rs"]
-cca-attester = ["tsm-report"]
-se-attester = ["pv"]
-tpm-attester = ["tss-esapi", "picky-asn1", "picky-asn1-x509", "picky-asn1-der"]
+csv-attester = []
+hygon-dcu-attester = []
+hygon = []
+cca-attester = []
+se-attester = []
+tpm-attester = []
 
 
 bin = ["clap", "tracing-subscriber"]
-- 
2.0
